builtin/credential/ldap/path_config.go

```diff
@@ -106,6 +106,11 @@ Default: cn`,
 				Default:     "tls12",
 				Description: "Maximum TLS version to use. Accepted values are 'tls10', 'tls11' or 'tls12'. Defaults to 'tls12'",
 			},
+			"deny_null_bind": &framework.FieldSchema{
+				Type:		 framework.TypeBool,
+				Default:	 false,
```

Comment:
I think it makes sense to have this true by default, but this actually creates an upgrade issue. So my suggestion is: flip it all to allow null bind rather than deny null bind. The default can then be false, and in the upgrade case it will be false, which means not allowed.